### PR TITLE
Istio CNI : Disable CNI repair for ambient tests

### PR DESCRIPTION
**Pull request by adilmp1**
This pull request combines changes from #49207 and #49228 to address issues related to the ambient test suite and CNI repair feature.

Firstly, this PR resolves the issue reported in #49207, which was overlooked in the existing ambient tests due to the enabled CNI repair feature. By disabling the CNI repair feature for integration tests, we ensure that the issue is properly identified and addressed. This approach enhances the reliability of our test coverage across both sidecar and ambient pods in the same namespace.

Additionally, this PR introduces a test to verify that enabling repair does not interfere with ambient in-pod configuration, as requested in #49228. A new test in integration/ambient/cnirepair/main_test.go sets repair to ON and validates that the system operates as expected. The ambient test suite now defaults to repair-OFF, with a dedicated test validating functionality when repair is set to ON.

### CODE REVIEW

**Comment by bleggett** [comment]
Hum. I'm not wild about disabling repair for _all_ the tests, since I would like to have some tests with it enabled, and some tests with it disabled.

But if the framework assumes a consistent install config for the entire suite, we're a bit limited unless we want to separate suites.

We have talked in the past about splitting the sidecar stuff out of the ambient suite, which would probably result in 

- ambient-only suite
- sidecar-only suite
- ambient-and-sidecar suite

 Even though I don't like disabling it, there's a strong argument to be made that _if_ repair kicks in in this suite, it's a bug - we should never need to repair stuff we are creating in integ suites.

File: tests/integration/ambient/cnirepair/main_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+var (
+	i istio.Instance
+
+	// Below are various preconfigured echo deployments. Whenever possible, tests should utilize these
+	// to avoid excessive creation/tear down of deployments. In general, a test should only deploy echo if
+	// its doing something unique to that specific test.
+	apps = &EchoDeployments{}
+
+	// used to validate telemetry in-cluster
+	prom prometheus.Instance
```
Comment (kfaseela) [inline tests/integration/ambient/cnirepair/main_test.go:50]: i think we are not using this

File: tests/integration/ambient/cnirepair/main_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+const (
+	WorkloadAddressedWaypoint = "workload-addressed-waypoint"
+	ServiceAddressedWaypoint  = "service-addressed-waypoint"
+	Captured                  = "captured"
+	Uncaptured                = "uncaptured"
+	SidecarWaypoint           = "sidecar-waypoint"
+	SidecarCaptured           = "sidecar-captured"
+	SidecarUncaptured         = "sidecar-uncaptured"
```
Comment (kfaseela) [inline tests/integration/ambient/cnirepair/main_test.go:127]: remove unused

**Review by howardjohn — COMMENTED** [review]
@bleggett do you think this is worth the cost (~2-3 minutes added to the test which currently takes about 8 minutes)?

**Comment by bleggett** [comment]
> @bleggett do you think this is worth the cost (~2-3 minutes added to the test which currently takes about 8 minutes)?

I don't love the added time cost, but we need to test both scenarios with at least a smoke, and currently the only way to test N different CNI configuration permutations in integ tests is to reinstall everything N different times. If the choice is between not validating that pod networking works, or paying the time cost, I would always prefer to pay the time cost.

Normally I would prefer unit tests but given that the CNI agent is effectively bifurcated into two parallel impls for sidecar and ambient from the jump, that's a bit tricky, as https://github.com/istio/istio/pull/49207 obviously showed.

tl;dr - I would rather pay the time cost now, and get coverage, then find a way to cut the time cost later, versus have no coverage.

**Comment by kfaseela** [comment]
Changes LGTM. Leaving to @bleggett for his review and approval

**Review by bleggett — APPROVED** [review]
LGTM. TY @adilmp1!

**Review by kfaseela — APPROVED** [review]
